### PR TITLE
Add Catppucin color styles

### PR DESCRIPTION
**Pull request by Stephen-Zulauf**
Added a Catppuccin theme to the demo style.c and overview.c. Used style guide located here: https://catppuccin.com/palette.

### CODE REVIEW

File: demo/common/overview.c [hunk 1]
```diff
@@ -156,7 +156,7 @@ overview(struct nk_context *ctx)
             nk_checkbox_flags_label(ctx, "No Scrollbar", &window_flags, NK_WINDOW_NO_SCROLLBAR);
             nk_checkbox_flags_label(ctx, "Minimizable", &window_flags, NK_WINDOW_MINIMIZABLE);
             nk_checkbox_flags_label(ctx, "Scale Left", &window_flags, NK_WINDOW_SCALE_LEFT);
-			nk_checkbox_label(ctx, "Disable widgets", &disable_widgets);
+        		nk_checkbox_label(ctx, "Disable widgets", &disable_widgets);
```
Comment (RobLoach) [inline demo/common/overview.c:159]:
``` suggestion
            nk_checkbox_label(ctx, "Disable widgets", &disable_widgets);
```

**Comment by RobLoach** [comment]
Thanks! Looks like there are 4 different schemes that they provide. Do you have plans to add all four?

**Comment by Stephen-Zulauf** [comment]
I can do that if you think it would be cool. :)

**Comment by Stephen-Zulauf** [comment]
I've added the four 'flavors' of Catppuccin described in the style guide here: https://catppuccin.com/palette. I left the entire pallet for each flavor in the form of nk_color structs for easy switching/customization. Wasn't sure how to label them in the drop-down on the overview demo, so I just abbreviated Catppuccin.

**Comment by Stephen-Zulauf** [comment]
awesome! sorry for the little mess I thought I caught everything.

**Pull request merged**